syntax = "proto3";

message Entry {
    int term = 1;
    string command = 2;
}

message AppendEntriesRequest() {
    int term = 1;
    int leaderId = 2;
    int prevLogIndex = 3;
    int prevLogTerm = 4;
    repeated Entry entries = 5;
    int leaderCommit = 6;
}

message AppendEntriesResponse() {
    int term = 1;
    bool success = 2;
}

message RequestVoteRequest() {
    int term = 1;
    int candidateId = 2;
    int lastLogIndex = 3;
    int lastLogTerm = 4;
}

message AppendEntriesResponse() {
    int term = 1;
    bool voteGranted = 2;
}


